Sources/Track.swift

```diff
@@ -32,28 +32,30 @@ class Track {
     }
     
     func track(event: String?,
+               eventID: TimedEventID?,
                properties: Properties? = nil,
-               timedEvents: InternalProperties,
+               timedEvents: TimedEvents,
                superProperties: InternalProperties,
                mixpanelIdentity: MixpanelIdentity,
-               epochInterval: Double) -> InternalProperties {
-        var ev = "mp_event"
-        if let event = event {
-            ev = event
+               epochInterval: Double) -> TimedEvents {
+        var eventName = "mp_event"
+        if let event {
+            eventName = event
         } else {
             Logger.info(message: "mixpanel track called with empty event parameter. using 'mp_event'")
         }
-        if !(mixpanelInstance?.trackAutomaticEventsEnabled ?? false) && ev.hasPrefix("$ae_") {
+        if !(mixpanelInstance?.trackAutomaticEventsEnabled ?? false) && eventName.hasPrefix("$ae_") {
             return timedEvents
         }
+        let timedEventID = eventID ?? eventName
```

Comment:
If no eventID is provided we fall back to using whatever the eventName is which means we default back to the original behavior.